core/src/test/scala/org/apache/spark/scheduler/OutputCommitCoordinatorSuite.scala

```diff
@@ -107,7 +107,7 @@ class OutputCommitCoordinatorSuite extends SparkFunSuite with BeforeAndAfter {
       val taskSet = invoke.getArguments()(0).asInstanceOf[TaskSet]
       new TaskSetManager(mockTaskScheduler, taskSet, 4) {
         private var hasDequeuedSpeculatedTask = false
-        override def dequeueSpeculativeTask(execId: String,
+        def dequeueSpeculativeTask(execId: String,
```

Comment:
sorry I hadn't noticed this before -- this change is wrong as now this whole setup isn't doing anything.  To keep the same behavior as before, I think you need to change `dequeueTaskHelper` to be `protected`, and then override it here.

https://github.com/squito/spark/commit/366e916f4e7499d477d6fafda538d5fe7d65d6a4

Reply:
Makes sense, have updated the code. Sorry, I missed out that one earlier. Thanks @squito.